datadog-log-ffi/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use datadog_log::logger;
+use datadog_log::logger::{logger_configure, logger_set_log_level};
+use ddcommon_ffi::{CharSlice, Error};
+
+/// Sets the global log level.
+///
+/// # Arguments
+/// * `log_level` - The minimum level for events to be logged
+///
+/// # Errors
+/// Returns an error if the log level cannot be set.
+#[no_mangle]
+pub extern "C" fn ddog_logger_set_log_level(
```

Comment:
I would check that log_level is between the proper range and return an error accordingly. C++ compilers will render an error if the value passed is not within the range of the enum but, unfortunately, with gcc is perfectly possible pass any value as long is an integer.

Reply:
I consider it a developer error if out of range values are passed, but I have still added error handling